### PR TITLE
[CI] Fix manual triggering of SYCL E2E workflow

### PR DESCRIPTION
**Pull request by uditagarwal97**
Currently, if we try to manually run SYCL E2E workflow, it does not execute any job. For example: https://github.com/intel/llvm/actions/runs/11062297137
This PR fixes that.

### CODE REVIEW

File: .github/workflows/sycl-linux-run-tests.yml [hunk 1]
```diff
@@ -146,6 +146,13 @@ on:
           - false
           - true
 
+      skip_run:
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:149]: I think line 161 should be fixed instead. Maybe `{{inputs.skip_run && inputs.skip_run == 'false'}}`, or `!= true`.

Comment (uditagarwal97) [reply .github/workflows/sycl-linux-run-tests.yml]: How about `if: github.event_name == 'workflow_dispatch' || inputs.skip_run == 'false'` ? Just run the E2E tests if we manually trigger the workflow.

**Review by aelovikov-intel — APPROVED** [review]

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers the PR is ready to be merged. E2E test failure on Gen12 Linux is unrelated.

**Comment by sarnex** [comment]
Fails not related

**Pull request merged**